### DISCUSSION TITLE
Feature Request - Agent name enhancement

### DISCUSSION THREAD

**Discussion by Jegelewicz** [ArctosDB/arctos#5269] [Ideas]
**Is your feature request related to a problem? Please describe.**
It is frustrating to know an agent name such as Kieth Barker is in Arctos, but a search for that name at https://arctos.database.museum/agents.cfm results in "Nothing Matched" 
![image](https://user-images.githubusercontent.com/5725767/156080070-47923333-7ee9-43a9-ab84-d35af4243d88.png)

when he is in Arctos, just as F. Kieth Barker

**Describe what you're trying to accomplish** 
Help people find the agents they are looking for

**Describe the solution you'd like**
I believe that I can add the aka Kieth Barker to this agent and then this will work, but that only fixes this one agent. Could we do a little agent name magic so that any preferred name that includes a first middle and last (3 names) gets an aka added? Maybe start with these two scenarios?

1. Any preferred name with a first name ending with a period gets and aka of second and third name
2. Any referred name with a second name ending with a period gets an aka of first and third name

I'm happy to work up a bulkload of akas if I can get the sql to narrow down the list of who needs them

Maybe this could also be a low quality agent report for the @ArctosDB/agents-committee to review twice a year or something?

It was also suggested that instead of "Nothing matched" when no exact matches are found, Maybe we should say something like

"No exact matches, try searching last name only"

FWIW, the above might also be good for the public search? https://arctos.database.museum/agent.cfm

**Describe alternatives you've considered**
Adding these as I find them.

**Additional context**

**Priority**
Please assign a priority-label. Unprioritized issues gets sent into a black hole of despair.

**Comment by campmlc** [comment]
I strongly support this.

**Comment by dustymc** [comment]
We've been down that road, it makes a lot of messes but it would not be technically difficult to turn some sort of name-making magic back on.

Something less magical/involving humans sounds better, but that would require some kind of ongoing resources.

No number of alternate spellings will find the occasional and inevitable "Keeth Barker" (unexpected spelling, whether given or just a typo) - suggest following the [docs](https://handbook.arctosdb.org/best_practices/Agents.html) (and flag the duplicate that occasionally uncovers, please!):

>thoroughly search to be sure the agent is not already in Arctos. Use the “clear form” button to ensure that you aren’t accidentally limiting your search. Using only the “any part of any name” option, search for last name,
